### PR TITLE
Concatenation support

### PR DESCRIPTION
**Pull request by GordStephen**
A first pass at concatenation support - requires axis types and index values to match when joining. Feedback appreciated!

### CODE REVIEW

**Comment by GordStephen** [comment]
Any feedback on this?

File: src/cat.jl [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        i += 1
+    end #while
+    return allequal
+end #equalvalued
+
+sizes{T<:AxisArray}(As::T...) = tuple(zip(map(size, As)...)...)
+matchingdims{N,T<:AxisArray}(As::NTuple{N,T}) = all(equalvalued, sizes(As...))
+matchingdimsexcept{N,T<:AxisArray}(As::NTuple{N,T}, n::Int) = all(equalvalued, sizes(As[[1:n-1; n+1:end]]...))
+
+Base.cat{T<:AxisArray}(n::Int, As::T...) = if n <= ndims(As[1])
```
Comment (mbauman) [inline src/cat.jl]: Minor nit: I'd prefer the `function` form here.

File: src/cat.jl [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    matchingdimsexcept(As, n) || error("All non-concatenated axes must be identically-valued")
+    newaxis = Axis{axisnames(As[1])[n]}(vcat(map(A -> A.axes[n].val, As)...))
+    checkaxis(newaxis)
```
Comment (mbauman) [inline src/cat.jl]: Ah, so just to confirm — if you concatenate along a dimension where each array has a Range axis, this just calls `vcat`  on those ranges, returning a dense vector, yes?  That's a great choice — it allows further optimizations down the road but doesn't lock us into any particular behavior at this time.

**Comment by mbauman** [comment]
Sorry it's taken me so long to look through this!  Thanks for the ping.

**Comment by GordStephen** [comment]
No problem! I changed the function syntax.

File: src/cat.jl (hunk 2 above)
Comment (GordStephen) [inline src/cat.jl]: Yep, correct. Definitely room for improvement down the line, but it seems like a reasonable starting point.

File: src/cat.jl (hunk 1 above)
Comment (GordStephen) [inline src/cat.jl]: Fixed in https://github.com/mbauman/AxisArrays.jl/pull/26/commits/d5b9d24d743916af432b68df3ddf497b097ccd93

**Pull request merged**